### PR TITLE
Add Discord UI Buttons to Verify Error + Ban Approval

### PR DESCRIPTION
**Pull request by 0xEmma**
## Types of changes

What types of changes does your code introduce?
*Put an `x` in the boxes that apply.*

- [ ] Bugfix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected).
- [ ] Documentation Update (if none of the other choices applies).

## Proposed changes

Discord native buttons! No more running commands when you can just hit it

## Checklist

*Put an `x` in the boxes that apply.*

- [x] I have read and followed the [CONTRIBUTING.md](https://github.com/hackthebox/Hackster/blob/main/CONTRIBUTING.md)
  doc.
- [x] Lint and unit tests pass locally with my changes.
- [x] I have added the necessary documentation (if appropriate).

## Additional context

Add any other context or screenshots here.

### CODE REVIEW

**Review by dimoschi — CHANGES_REQUESTED** [review]
The PR is in the right direction, however the `helpers.ban.ban_member` already sends a message to the `SR_MOD` channel if approval is required, so maybe the Approve/Deny buttons should be added there.

**Comment by 0xEmma** [comment]
> 

Will make that change

**Comment by 0xEmma** [comment]
The code already sends it in the ban_member class, its just defined in the ban cog

File: src/cmds/core/ban.py [hunk 1]
```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    class BanApprove(discord.ui.view):
```
Comment (dimoschi) [inline src/cmds/core/ban.py:27]: `discord.ui.view` is not a class, so you cannot extend. Maybe you wanted to write `discord.ui.view.View`. In that case better import it `from discord.ui.view import View` and use it.

Comment (dimoschi) [reply src/cmds/core/ban.py:27]: Also, why BanApprove is a nested class? This is uncommon pattern in Python.

File: src/cmds/core/ban.py [hunk 2]
```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
...
+        def __init__(self, ban_id):
```
Comment (dimoschi) [inline src/cmds/core/ban.py:28]: Please add type annotation for `ban_id`

File: src/cmds/core/ban.py [hunk 3]
```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
...
+                    message = "Cannot find record of ban request. Has this user already been unbanned?"
+
+                ban_to_update.approved = True
+                await session.commit()
+
+                stmt = select(Ban).filter(Ban.id == self.ban_id)
+                result = await session.scalars(stmt)
+                ban: Ban = result.first()
+
+            member = await self.bot.get_member_or_user(interaction.guild, ban.user_id)
```
Comment (dimoschi) [inline src/cmds/core/ban.py:45]: Here `self` refers to the instance of the class `BanApprove`, which doesn't have access to `bot` member that exists in "parent class" `BanCog`

File: src/cmds/core/ban.py [hunk 4]
```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
...
+            if not member:
+                message = f"User {ban.user_id} not found."
+
+            self.bot.loop.create_task(
```
Comment (dimoschi) [inline src/cmds/core/ban.py:49]: Same as above `BanApprove` doesn't have access to `bot`

File: src/cmds/core/ban.py [hunk 5]
```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
...
+                schedule(unban_member(interaction.guild, member), run_at=datetime.fromtimestamp(ban.unban_time))
+            )
+            if not message:
+                message = "Ban Approval Recorded."
+            await interaction.response.send_message(message, ephemeral=True)
+            self.disable_all_items()
+
+        @discord.ui.button(label="Deny", style=discord.ButtonStyle.red)
+        async def deny_ban(self, interaction: discord.Interaction, button: discord.ui.Button):
```
Comment (dimoschi) [inline src/cmds/core/ban.py:58]: The `button` argument is not used, maybe remove it?

File: src/cmds/core/ban.py [hunk 6]
```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    class BanApprove(discord.ui.view):
+        def __init__(self, ban_id):
+            super().__init__(timeout=None)
+            self.ban_id = ban_id
+        @discord.ui.button(label="Approve", style=discord.ButtonStyle.primary)
+        async def approve_ban(self, interaction: discord.Interaction, button: discord.ui.Button):
```
Comment (dimoschi) [inline src/cmds/core/ban.py:32]: The `button` argument is not used, maybe remove it?

File: src/cmds/core/identify.py [hunk 7]
```diff
@@ -22,6 +22,37 @@ class IdentifyCog(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    class VerifyError(discord.ui.view):
+        def __init__(self, user_id):
+            super().__init__(timeout=None)
+            self.user_id = user_id
+
+        @discord.ui.button(label="Remove Token", style=discord.ButtonStyle.primary)
+        async def remove_Token(self, interaction: discord.Interaction, button: discord.ui.Button):
```
Comment (dimoschi) [inline src/cmds/core/identify.py:31]: Function names should be all lowercase.

Comment (dimoschi) [reply src/cmds/core/identify.py:31]: Also, `button` is not being used.

File: src/cmds/core/identify.py [hunk 8]
```diff
@@ -22,6 +22,37 @@ class IdentifyCog(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    class VerifyError(discord.ui.view):
```
Comment (dimoschi) [inline src/cmds/core/identify.py:25]: `discord.ui.view` is not a class, so you cannot extend. Maybe you wanted to write `discord.ui.view.View`. In that case better import it `from discord.ui.view import View` and use it.

**Review by dimoschi — CHANGES_REQUESTED** [review]
Many changes are required for this PR to work.

**Pull request closed without merging**